src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -123,6 +123,7 @@ public static async ValueTask<QuicConnection> ConnectQuicAsync(HttpRequestMessag
                     MaxInboundBidirectionalStreams = 0, // Client doesn't support inbound streams: https://www.rfc-editor.org/rfc/rfc9114.html#name-bidirectional-streams. An extension might change this.
                     MaxInboundUnidirectionalStreams = 5, // Minimum is 3: https://www.rfc-editor.org/rfc/rfc9114.html#unidirectional-streams (1x control stream + 2x QPACK). Set to 100 if/when support for PUSH streams is added.
                     IdleTimeout = idleTimeout,
+                    HandshakeTimeout = handshakeTimeout,
```

Comment:
Nit: this could just be `Timeout.InfiniteTimeSpan`

The cancellationToken passed to `ConnectAsync` is already backed by `ConnectTimeout`. Making the handshake timeout the same value may make it a coin flip whether you see a handshake timeout vs. connect timeout exception.